### PR TITLE
check filesystems array has no duplicate mountpoints or blockdevs

### PR DESCRIPTION
**Pull request by kwaegema**

### CODE REVIEW

File: quattor/filesystems.pan [hunk 1]
```diff
@@ -3,6 +3,26 @@
 }
 declaration template quattor/filesystems;
 
+function filesystems_uniq_paths = {
+    mounts = nlist();
```
Comment (jrha) [inline quattor/filesystems.pan]: `dict()`

File: quattor/filesystems.pan [hunk 2]
```diff
@@ -3,6 +3,26 @@
...
+    blockdevs = nlist();
```
Comment (jrha) [inline quattor/filesystems.pan]: `dict()`

**Comment by jrha** [comment]
Other than in-line comments, LGTM

**Comment by kwaegema** [comment]
@jrha done

File: quattor/filesystems.pan [hunk 3]
```diff
@@ -3,6 +3,26 @@
 }
 declaration template quattor/filesystems;
 
+function filesystems_uniq_paths = {
```
Comment (stdweird) [inline quattor/filesystems.pan]: needs documentation, esp on the `SELF` part (ie where is this called from)

**Comment by kwaegema** [comment]
@stdweird doc added

File: quattor/filesystems.pan [hunk 4]
```diff
@@ -3,6 +3,30 @@
 }
 declaration template quattor/filesystems;
 
+@documentation{ 
+    desc = check that no duplicate mountpoints or blockdevices are used
+    arg = array of structure_filesystem
```
Comment (stdweird) [inline quattor/filesystems.pan]: where is the argument used in the code?

Comment (kwaegema) [inline quattor/filesystems.pan]: @stdweird updated the commit

File: quattor/filesystems.pan [hunk 5]
```diff
@@ -3,6 +3,30 @@
 }
 declaration template quattor/filesystems;
 
+@documentation{ 
+    desc = check that no duplicate mountpoints or blockdevices are used
+    arg = array of structure_filesystem, from quattor/types/system
```
Comment (stdweird) [inline quattor/filesystems.pan]: but where is it used in the code below? there's only `SELF`, not `ARGV`

File: quattor/filesystems.pan [hunk 6]
```diff
@@ -3,6 +3,30 @@
...
+}
+function filesystems_uniq_paths = {
+    mounts = dict();
+    blockdevs = dict();
+    foreach (idx;data;SELF){
```
Comment (jrha) [inline quattor/filesystems.pan]: Should this not be `ARGV[0]` rather than `SELF`?

**Comment by kwaegema** [comment]
@jrha @stdweird indeed, i looked over it. fixed it now and rebased

**Pull request merged**